### ISSUE TITLE
Can't step through Python library files

### ISSUE THREAD

**Issue by hawkerm** [Microsoft/PTVS#106]
I was trying to debug a crash through the Python stack (I had to do this for a bizarre unittest error I was getting) the other day.

I could set a breakpoint in the CPython 2.7 library file that I dove into through the Call Stack (think I had to turn off debugging 'just my code' in the options first), but as soon as I tried to 'step' anywhere in any one of the Python libraries, I would suddenly jump back out past all the instructions to **my** next piece of code.  I had to keep setting breakpoints across the Python libraries to trace down what I wanted to look at and where I wanted to stop.

Certainly not expected.

**Comment by int19h** [comment]
This is by design - have a look at Tools -> Options -> Python Tools -> Debugging -> Enable debugging of the Python standard library. It's so that when you e.g. use a standard library function that takes a callback, you can step directly into it, instead of going through layers and layers of stdlib.

**Comment by hawkerm** [comment]
Thanks int19h!

Didn't know the Python Tools had options there, always forget to look.

Should be what I need, I'll try it out next time I hit an issue.

**Comment by hawkerm** [comment]
Hmm, so I enabled the debugger option and I'm trying to step into an 'assertAlmostEqual' call on my unittest.TestCase, but it just skips over it still.

**Comment by zooba** [comment]
It's entirely likely that we don't flow this option into the debugger when launching tests. I guess `ptvsd` should grow some additional options for setting up this kind of thing, but that's unlikely to happen for 2.2 at this stage.

As a workaround, if you have `unittest.main()` in your test file you should be able to right-click->Start With Debugging to get similar behaviour. Doing it this way will use regular debugging, rather than the tricks we need to play to get the test window to work for us.

**Comment by hawkerm** [comment]
Thanks for the tip, I'll try that out sometime.  My test cases are visual and take a while, so I was trying to debug a specific one from the Test Explorer.

**Comment by hawkerm** [comment]
Hmm, I have something weird going on with my packages, so the imports don't match up if I try and run the testcase from it's own file vs. when it's rooted in my project from Test Explorer.

**Comment by zooba** [comment]
Ah yeah, we end up running with different search path settings here :( I thought I mostly fixed those in 2.2 RC, but certainly we do some special things via Test Explorer that would generally not be appropriate when launching a script directly.

Without being able to see how your project is laid out, I can't make any specific suggestions on how to fix it, except that (1) paths are probably not correct, (2) it's not your fault, and (3) you can temporarily modify Search Paths through your project or `sys.path` in your script to fix it.

If you figure out exactly which paths aren't showing up, let us know and we'll take a look (for example, "my project's working directory property is being set wrong in sys.path").

**Comment by hawkerm** [comment]
Thanks, I'll try outputting sys.path in all cases and see how it differs next time and let you know.

If you're curious about where all these bugs I keep filing are coming from, my project's also on GitHub: [SpaceBattleArena](https://github.com/Mikeware/SpaceBattleArena/tree/master/SBA_Serv).

The tools have certainly made it a ton more manageable than the past, and helped me even develop my test cases and be much more efficient until I run into some of these odd complex cases...

**Comment by Oceania2018** [comment]
It works when I enable these two options in Visual Studio 2017.
Now I can step into numpy package.

![image](https://user-images.githubusercontent.com/1705364/50709770-82c50200-102e-11e9-962e-047827295e96.png)

**Comment by zwn** [comment]
I can confirm that the option "Use legacy debugger" needs to be checked. Should a new issue be open for this?

**Comment by KamilGadawski** [comment]
> 
> 
> It works when I enable these two options in Visual Studio 2017.
> Now I can step into numpy package.
> 
> ![image](https://user-images.githubusercontent.com/1705364/50709770-82c50200-102e-11e9-962e-047827295e96.png)

If I select these two options, I get an error 

zipimport.ZipImportError: not a Zip file

Only after the Ude legacy debugger is awarded, the program starts, but the debugger still does not work. How do I fix it? I use VS Community 2019
